File: AdministracionDeTienda/src/gestorAplicacion/clasesBase/Venta.java
package gestorAplicacion.clasesBase;

import java.io.Serializable;
import java.util.ArrayList;

import gestorAplicacion.clasesHerencia.Transportista;

public class Venta implements Serializable{

	/**
	 *
	 */
	private static final long serialVersionUID = 1L;
	private ArrayList<Producto> productosVenta;
	private double total;
	private Object comprador;
	private OfertaPorDefecto productosOfertados;
	private double ganancias;
	private static double porcentajeBanco;
	private Transportista repartidor;

	public Venta(Object comprador) {

		this.comprador=comprador;

	}

	public Venta(Object comprador, double total, ArrayList<Producto> productosVenta, OfertaPorDefecto productosOfertados, Transportista repartidor) {
		this(comprador);
		this.total=total;
		this.productosVenta=productosVenta;
		this.setProductosOfertados(productosOfertados);
		this.repartidor=repartidor;
	}

	public static void asignarPorcentajeBanco(Venta v) {
		if(porcentajeBanco!=0){
			Tienda.getCuentaTienda().getEntidad().abonarCuentaAuxiliar(porcentajeBanco*v.ganancias);
			Tienda.getCuentaTienda().setDinero(Tienda.getCuentaTienda().getDinero()-porcentajeBanco*v.ganancias);
		}
	}

	public double getTotal() {
		return total;
	}

	public void setTotal(double total) {
		this.total = total;
	}


	public ArrayList<Producto> getProductosVenta() {
		return productosVenta;
	}

	public void setProductosVenta(ArrayList<Producto> productosVenta) {
		this.productosVenta = productosVenta;
	}

	public Object getComprador() {
		return comprador;
	}

	public void setComprador(Object comprador) {
		this.comprador = comprador;
	}

	public OfertaPorDefecto getProductosOfertados() {
		return productosOfertados;
	}

	public void setProductosOfertados(OfertaPorDefecto productosOfertados) {
		this.productosOfertados = productosOfertados;
	}

	public double getGanancias() {
		return ganancias;
	}

	public void setGanancias(double ganancias) {
		Tienda.getCuentaTienda().setDinero(ganancias+Tienda.getCuentaTienda().getDinero());
		this.ganancias = ganancias;
	}

	public Transportista getRepartidor() {
		return repartidor;
	}

	public void setRepartidor(Transportista repartidor) {
		this.repartidor = repartidor;
	}

	public static double getPorcentajeBanco() {
		return porcentajeBanco;
	}

	public static void setPorcentajeBanco(double porcentajeBanco) {
		Venta.porcentajeBanco = porcentajeBanco;
	}

	@Override
	public String toString() {
		String s="";
		try {
		for(Producto p: productosVenta) {
			s+=p.toString();
		}
		return String.format("\n\nProductos vendidos: %s\n\nTotal de la venta: $%g", s,total) ;
		}
		catch(NullPointerException e) {
			return "\n\n...No se encontraron productos\npara completar la venta\n\n";
		}
	}

}

File: Python/src/gestorAplicacion/clasesBase/Venta.py
import pickle


class Venta:
    porcentajeBanco = 0

    def __init__(self, comprador, productosOfertados = None, repartidor = None):
        self.productosVenta = []
        self.total = 0
        self.comprador = comprador
        self.productosOfertados = productosOfertados
        self.ganancias = 0
        self.repartidor = repartidor

    @classmethod
    def asignarPorcentajeBanco(cls,venta):
        if cls.porcentajeBanco != 0:
            Tienda.getCuentaTienda().getEntidad().abonarCuentaAuxiliar(
                cls.porcentajeBanco * venta.getGanancias()
            )
            Tienda.getCuentaTienda().setDinero(
                Tienda.getCuentaTienda().getDinero() - cls.porcentajeBanco * venta.getGanancias()
            )

    def getTotal(self):
        return self.total

    def setTotal(self, total):
        self.total = total

    def getProductosVenta(self):
        return self.productosVenta

    def setProductosVenta(self, productosVenta):
        self.productosVenta = productosVenta

    def getComprador(self):
        return self.comprador

    def setComprador(self, comprador):
        self.comprador = comprador

    def getProductosOfertados(self):
        return self.productos_ofertados

    def setProductosOfertados(self, productosOfertados):
        self.productosOfertados = productosOfertados

    def getGanancias(self):
        return self.ganancias

    def setGanancias(self, ganancias):
        CuentaTienda.getCuentaTienda().setDinero(ganancias + CuentaTienda.getCuentaTienda().getDinero())
        self.ganancias = ganancias

    def getRepartidor(self):
        return self.repartidor

    def setRepartidor(self, repartidor):
        self.repartidor = repartidor

    @classmethod
    def getPorcentajeBanco(cls):
        return cls.porcentajeBanco

    @classmethod
    def setPorcentajeBanco(cls, porcentajeBanco):
        cls.porcentajeBanco = porcentajeBanco

    def __str__(self):
        s = ""
        try:
            for producto in self.productos_venta:
                s += str(producto)
            return f"\n\nProductos vendidos: {s}\n\nTotal de la venta: ${self.total}"
        except AttributeError:
            return "\n\n...No se encontraron productos\npara completar la venta\n\n"
